test_board.py

```diff
@@ -23,48 +32,123 @@ def test_when_a_user_picks_a_spot_board_is_updated_with_their_spot():
+                                                            [2, 5, 8]
+                                                         ]
+def test_win_by_column_returns_possible_LTR_diagonal_win():
+    win_config = WinningCombos(3)
+    win_config.diagonal_left_to_right_win(win_config.win_by_row()) == [0, 4, 8]
+
+def test_win_by_column_returns_possible_RTL_diagonal_win():
+    win_config = WinningCombos(3)
+    win_config.diagonal_right_to_left_win(win_config.win_by_row()) == [2, 4, 6]
+
+def test_winning_combos_are_created_for_3x3_board():
+    win_config = WinningCombos(3)
+    win_config.create_winning_combos()
+    assert win_config.winning_combos == [
+                                         [0, 1, 2],
+                                         [3, 4, 5],
+                                         [6, 7, 8],
+                                         [0, 3, 6],
+                                         [1, 4, 7],
+                                         [2, 5, 8],
+                                         [0, 4, 8],
+                                         [2, 4, 6],
+                                        ]
+    
+def test_game_ends_if_user_occupies_3_in_column():
+    win_config = WinningCombos(3)
+    win_config.create_winning_combos()
     game_board = Board()
-    game_board.new_game()
-    assert display_board(game_board.board_state) == game_board.board_state[0] + " | " + game_board.board_state[1] + " | " + game_board.board_state[2] + "\n" +"=========" + "\n" + game_board.board_state[3] + " | " + game_board.board_state[4] + " | " + game_board.board_state[5] + "\n" + "=========" + "\n" + game_board.board_state[6] + " | " + game_board.board_state[7] + " | " + game_board.board_state[8] 
+    game_board.board_state = [
+                                "X","O","X",
+                                "X","","",
+                                "X","","X"
+                             ]
+    assert EndStates.did_a_player_win(game_board.board_state, 'X', win_config.winning_combos) == True
 
+def test_game_ends_if_user_occupies_3_in_row():
+    win_config = WinningCombos(3)
+    win_config.create_winning_combos()
+    game_board = Board()
+    game_board.board_state = [
+                                "X","X","X",
+                                "O","","",
+                                "O","","X"
+                             ]
+    assert EndStates.did_a_player_win(game_board.board_state, 'X', win_config.winning_combos) == True
+    
+def test_game_ends_if_user_occupies_3_in_diagonal():
+    win_config = WinningCombos(3)
+    win_config.create_winning_combos()
+    game_board = Board()
+    game_board.board_state = [
+                                "X","O","O",
+                                "O","X","",
+                                "O","","X"
+                             ]
+    assert EndStates.did_a_player_win(game_board.board_state, 'X', win_config.winning_combos) == True
```

Comment:
Overall, this is the best suite of tests I've seen from you so far. Kudos 👏

Reply:
Thank you! Most of my time has been spent with the board.